src/main/java/Parser.java

```diff
@@ -0,0 +1,105 @@
+/**
+ * The {@code Parser} class processes and parses user input, coming up with a response.
+ */
+public class Parser {
+
+    /**
+     * Initialises Parser object.
+     */
+    public Parser() {
+    }
+
+    /**
+     * Processes user input and returns an appropriate reply.
+     *
+     * @param lst The {@code TaskList} that keeps track of all pending tasks.
+     * @param msg User input.
+     * @return Response to user input.
+     * @throws JimmyException If user input contains an error.
+     */
+    public static String parse(TaskList lst, String msg) throws JimmyException {
+
+        String[] arr = msg.split(" ");
+        boolean isLong = arr.length > 1;
+        String reply = "";
+
+        if (!isLong) {
```

Comment:
Maybe can consider swap your if and else block to "make the happy path prominent" according to textbook?

Reply:
Hmm interesting, I will look into it. Thanks for input.